driver-async/src/test/functional/com/mongodb/async/client/JsonPoweredCrudTestHelper.java

```diff
@@ -247,6 +263,44 @@ BsonDocument getAggregateResult(final BsonDocument collectionOptions, final Bson
         return toResult(iterable);
     }
 
+    BsonDocument getDatabaseAggregateResult(final BsonDocument collectionOptions, final BsonDocument arguments,
+                                            @Nullable final ClientSession clientSession) {
+        assumeTrue(!isSharded());
+        List<BsonDocument> pipeline = new ArrayList<BsonDocument>();
+        for (BsonValue stage : arguments.getArray("pipeline")) {
+            pipeline.add(stage.asDocument());
+        }
+
+        AggregateIterable<BsonDocument> iterable;
+        if (clientSession == null) {
+            iterable = database.aggregate(pipeline, BsonDocument.class);
+        } else {
+            iterable = database.aggregate(clientSession, pipeline, BsonDocument.class);
+        }
+
+        if (arguments.containsKey("allowDiskUse")) {
+            iterable.allowDiskUse(arguments.getBoolean("allowDiskUse").getValue());
+        }
+        if (arguments.containsKey("batchSize")) {
+            iterable.batchSize(arguments.getNumber("batchSize").intValue());
+        }
+        if (arguments.containsKey("collation")) {
+            iterable.collation(getCollation(arguments.getDocument("collation")));
+        }
+
+        BsonDocument results = toResult(iterable);
+        for (BsonValue result : results.getArray("result", new BsonArray())) {
```

Comment:
On direct connections the driver adds readPreference primaryPreferred to commands that don't have a set readPref. This also removes other metadata ($clusterTime, signature & keyId) so that from the results of the aggregation can be comparable.